cmd/skopeo/sync.go

```diff
@@ -0,0 +1,594 @@
+				if err != nil {
+					logrus.WithFields(logrus.Fields{
+						"repo":     imageName,
+						"registry": server,
+					}).Error("Error processing repo, skipping")
+					logrus.Error(err)
+					continue
+				}
+
+				sourceReferences, err = imagesToCopyFromRegistry(imageRef, repoName, serverCtx)
+				if err != nil {
+					logrus.WithFields(logrus.Fields{
+						"repo":     imageName,
+						"registry": server,
+					}).Error("Error processing repo, skipping")
+					logrus.Error(err)
+					continue
+				}
+			}
+
+			if len(sourceReferences) == 0 {
+				logrus.WithFields(logrus.Fields{
+					"repo":     imageName,
+					"registry": server,
+				}).Warnf("No tags to sync found")
+				continue
+			}
+			repoDescList = append(repoDescList, repoDescriptor{
+				TaggedImages: sourceReferences,
+				Context:      serverCtx})
+		}
+	}
+
+	return
+}
+
+func (opts *syncOptions) run(args []string, stdout io.Writer) error {
+	if len(args) != 2 {
+		return errorShouldDisplayUsage{errors.New("Exactly two arguments expected")}
+	}
+
+	policyContext, err := opts.global.getPolicyContext()
+	if err != nil {
+		return errors.WithMessage(err, "Error loading trust policy")
+	}
+	defer policyContext.Destroy()
+
+	destinationURL, err := parseURL(args[1])
+	if err != nil {
+		return errors.WithMessage(err, "Error while parsing destination")
+	}
+	destinationCtx, err := opts.destImage.newSystemContext()
+	if err != nil {
+		return err
+	}
+
+	sourceCtx, err := opts.srcImage.newSystemContext()
+	if err != nil {
+		return err
+	}
+	sourceArg := args[0]
+
+	var srcRepoList []repoDescriptor
+
+	if opts.sourceYaml {
+		srcRepoList, err = syncFromYaml(sourceArg, sourceCtx)
+		if err != nil {
+			return err
+		}
+	} else {
+		sourceURL, err := parseURL(sourceArg)
+		if err != nil {
+			return errors.WithMessage(err, "Error while parsing source")
+		}
+
+		if transports.Get(sourceURL.Scheme) == directory.Transport &&
+			sourceURL.Scheme == destinationURL.Scheme {
+			return errors.New("sync from 'dir:' to 'dir:' not implemented, use something like rsync instead")
+		}
+
+		srcRepo, err := syncFromURL(sourceURL, sourceCtx)
+		if err != nil {
+			return err
+		}
+		srcRepoList = append(srcRepoList, srcRepo)
+	}
+
+	ctx, cancel := opts.global.commandTimeoutContext()
+	defer cancel()
+
+	var imgCounter int
+	for _, srcRepo := range srcRepoList {
+		options := copy.Options{
+			RemoveSignatures: opts.removeSignatures,
+			SignBy:           opts.signByFingerprint,
+			ReportWriter:     os.Stdout,
+			DestinationCtx:   destinationCtx,
+			SourceCtx:        srcRepo.Context,
+		}
+
+		for counter, ref := range srcRepo.TaggedImages {
+			destRef, err := buildFinalDestination(ref, destinationURL, srcRepo.DirBasePath)
+			if err != nil {
+				return err
+			}
+
+			logrus.WithFields(logrus.Fields{
+				"from": transports.ImageName(ref),
+				"to":   transports.ImageName(destRef),
+			}).Infof("Copying image tag %d/%d", counter+1, len(srcRepo.TaggedImages))
```

Comment:
All Infof logs are printed by default which does not look right to me. I suggest printing those logs on stdout.